### PR TITLE
add inline guid processing

### PR DESCRIPTION
**Pull request by HEskandari**
Scrubs the guid values within a string

### CODE REVIEW

**Comment by SimonCropp** [comment]
can u pull in the upstream. you are a few commits behind

**Comment by HEskandari** [comment]
@SimonCropp done

File: src/Verify.Tests/Serialization/SerializationTests.ShouldScrubInlineGuidsByDefault.verified.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  Title: item Guid_1 - (ID={Guid_1}),
+  Variant: {
+    Id: variant id: Guid_2
```
Comment (SimonCropp) [inline src/Verify.Tests/Serialization/SerializationTests.ShouldScrubInlineGuidsByDefault.verified.txt:4]: why have the single quotes around the text gone?

**Comment by SimonCropp** [comment]
can you add some doco for this feature

File: src/Verify.Tests/Serialization/SerializationTests.ShouldScrubInlineGuidsByDefault.verified.txt (hunk 1 above)
Comment (HEskandari) [reply src/Verify.Tests/Serialization/SerializationTests.ShouldScrubInlineGuidsByDefault.verified.txt:4]: Good catch. Thing is that up the stack there is a code that does this:

```
if (sharedScrubber.TryConvertString(valueAsString, out var result))
{
    writer.WriteRawValue(result);
    return;
}
```

The `WriteRawValues` expects you to have quotes in place. Changing it to `WriteValues` puts the quotes for all the values. I'll send an update that'd put in the quotes manually when formatting the string.

**Comment by HEskandari** [comment]
@SimonCropp added some snippets.

**Comment by SimonCropp** [comment]
This is now deployed. NuGet may take some time to make it available for download.

note i made it an opt in feature for backwards compat reasons

**Pull request merged**